src/Components/WebAssembly/WebAssembly/src/Hosting/RootComponentTypeCache.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    internal static class RootComponentTypeCache
```

Comment:
This is the same as ServerComponentTypeCache but without the concurrency concerns

Reply:
@SteveSandersonMS had some concerns about assuming behavior around concurrency. If it's cheap, can we just apply the same concerns as server?

Reply:
It's using the same cache as the server now